config/action.d/nftables-common.conf

```diff
@@ -0,0 +1,96 @@
+# Fail2Ban configuration file
+#
+# Author: Daniel Black
+# Author: Cyril Jaquier
+# Modified: Yaroslav O. Halchenko <debian@onerussian.com>
+# 			made active on all ports from original iptables.conf
+# Modified: Alexander Belykh <albel727@ngs.ru>
+#                       adapted for nftables
+#
+# This is a included configuration file and includes the definitions for the nftables
+# used in all nftables based actions by default.
+#
+# The user can override the defaults in nftables-common.local
+
+[INCLUDES]
+
+after = nftables-common.local
+
+[Definition]
+
+# Option:  nftables_mode
+# Notes.:  additional expressions for nftables filter rule
+# Values:  nftables expressions
+#
+nftables_mode = <protocol> dport { <port> }
+
+# Option:  actionstart
+# Notes.:  command executed once at the start of Fail2Ban.
+# Values:  CMD
+#
+actionstart = <nftables> add set filter f2b-<name> { type ipv4_addr\; }
+              <nftables> insert rule filter <chain> %(nftables_mode)s ip saddr @f2b-<name> <blocktype>
+
+# Option:  actionstop
+# Notes.:  command executed once at the end of Fail2Ban
+# Values:  CMD
+#
+actionstop = HANDLE_ID=$(<nftables> --handle --numeric list chain filter <chain> | grep -m1 'ip saddr @f2b-<name> <blocktype> # handle' | grep -oe ' handle [0-9]*'); <nftables> delete rule filter <chain> $HANDLE_ID
```

Comment:
this is bloody pity if with nft we need this kind of contraption to clean things up after ourselves :-/  great to learn about "-o" for grep though! ;)
if there is no way around somehow, we should at least split this into some helper variables and use string interpolations, e.g.

```
_nft_list = <nftables> --handle --numeric list chain filter <chain>
_get_handle_id = grep -m1 'ip saddr @f2b-<name> <blocktype> # handle' | grep -oe ' handle [0-9]*'

actionstop = HANDLE_ID=$(%(_nft_list)s | %(_get_handle_id)s); <nftables> delete rule filter <chain> $HANDLE_ID
```
so it fits the screen somewhat better